src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -393,12 +393,12 @@ public void improveRecipe(IRecipeStorage recipe, int count, ICitizenData citizen
                 {
                     reducedItem = input.getItemStack();
                     reducedItem.setCount(input.getAmount() - 1);
-                    newRecipe.add(reducedItem);
+                    newRecipe.add(new ItemStorage(reducedItem));
                     didReduction = true;
                 }
                 else
                 {
-                    newRecipe.add(input.getItemStack());
+                    newRecipe.add(new ItemStorage(input.getItemStack()));
```

Comment:
Why does this take an ItemStorage, convert it to an ItemStack, then back to an ItemStorage?  Shouldn't you just be copying the ItemStorage with its existing settings intact?

Reply:
Yeah true, this should probably make a real copy

Reply:
There might be other places too (I didn't look very hard); probably should be suspicious of anywhere that uses `getItemStack` to ensure it's not doing this sort of pattern.

Reply:
This is likely the only place this is happening. We don't have a copy on ItemStorage today. That having been said, I'll add one, since future changes will require it.